Makefile

```diff
@@ -0,0 +1,114 @@
+# Based on the work of Thomas Poignant (thomaspoignant)
+# https://gist.github.com/thomaspoignant/5b72d579bd5f311904d973652180c705
+GOCMD=go
+GOTEST=$(GOCMD) test
+GOVET=$(GOCMD) vet
+BINARY_NAME=kluctl
+TEST_BINARY_NAME=kluctl-e2e
+REQUIRED_ENV_VARS=GOOS GOARCH
+EXPORT_RESULT?=false
+
+GREEN  := $(shell tput -Txterm setaf 2)
+YELLOW := $(shell tput -Txterm setaf 3)
+WHITE  := $(shell tput -Txterm setaf 7)
+CYAN   := $(shell tput -Txterm setaf 6)
+RESET  := $(shell tput -Txterm sgr0)
+
+.PHONY: all test build vendor check-env check-kubectl check-helm check-kind
+
+all: help
+
+## Check:
+check-env: ## Checks if required environment variables are set
```

Comment:
I would prefer to not make these mandatory, which would then allow us to simply build for the local machine as target.

Reply:
Do you want to add different targets for linux, windows, darwin, amd64 (e.g. https://github.com/awslabs/amazon-eks-ami/blob/aed6c88d7783464edb443d3e1c796ddce2a30375/Makefile#L46) .... or some sort of "magic" for setting GOOS and GOARCH?

Reply:
The default behaviour of golang is to use the local os/arch as default if GOOS/GOARCH is unset or empty, so you can simply remove the check and then let it do it's default thingy.

Having extra targets for all os/arch combinations is then a different topic, but I assume it's a good idea to have it.

Reply:
Yes, but we need it for the python target (GOARCH). Alternative solution: We download it for all archs?

Reply:
I removed GOARCH and GOOS variables and added the download of python for all archs.

Reply:
The arch argument to the download-python.sh script is optional. It will internally default to use `uname -s`.

Reply:
Okay, but then its not working on Windows, right?

Reply:
It works when used with MinGW (I tested this a few weeks ago). If we want to get rid of the MinGW requirement, we'd have to get rid of bash scripts nevertheless so I think it's fine to rely on MinGW.

Reply:
Changed it